### ISSUE TITLE
Feature request: DKIM authentication setup

### ISSUE THREAD

**Issue by jleclanche** [hashicorp/terraform-provider-googleworkspace#138]
Configuration can be found here: https://admin.google.com/ac/apps/gmail/authenticateemail

Suggested configuration:

```hcl
resource "googleworkspace_dkim_authentication_key "example" {
  domain_name = "example.com"
  prefix = "google" // default
  key_size = 2048 // default
}

// Example usage in Cloudflare DNS config
resource "cloudflare_record" "google_dkim_example" {
  zone_id = cloudflare_zone.example.id
  name    = googleworkspace_dkim_authentication_key.example.record_name // "google._domainkey"
  type    = googleworkspace_dkim_authentication_key.example.record_type // "TXT"
  value    = googleworkspace_dkim_authentication_key.example.record_value // "v=DKIM1; k=rsa; p=…"
}
```

**Comment by megan07** [comment]
Hi @jleclanche! Thanks for reaching out and submitting this request. I see where I can update this in the UI, however, I can't find an exposed API for this. If you're aware of one, could you please share it? Thanks!

I browsed through here and didn't see anything, but I could have missed it! https://developers.google.com/admin-sdk

**Comment by jleclanche** [comment]
@megan07 you're right I can't find it in the docs either. I took a look at the console just to see if it could give me pointers but it looks like it's using a weird rpc (i thought it was protobuf at first but no) over a service called `GmailAdminSettingsUi`.

https://admin.google.com/_/GmailAdminSettingsUi/data/batchexecute?rpcids=ljYEJ&f.sid=1551415815315915618&bl=boq_gmail-adminsettings-ui_20210815.02_p0&hl=en&cid=02f57xck&_reqid=978009&rt=c

It looks like it would be mentioned here if available: https://developers.google.com/admin-sdk/admin-settings

I don't suppose we can reach out to Google to see if they can expose an API for this?

**Comment by megan07** [comment]
@jleclanche - a quick search and I wonder if [this](https://issuetracker.google.com/savedsearches/566264) would be the place to create that feature request.

We've reached out a couple times internally around other issues and have not heard back, but I'll have @tejavar reach out again.

Thanks!
